File: stack/js/stack.js
class Stack {
  constructor() {
    this.stack = [];
    this.top = -1;
  }

  addElement = (item) => {
    this.stack.push(item);
    this.top += 1;
  }

  checkIfEmpty = () => {
    return this.stack.length === 0;
  }

  removeElement = () => {
    if (this.checkIfEmpty === true) {
      return;
    }
    this.stack.pop();
    this.top -= 1;
  }

  peek = () => {
    return this.stack[this.top];
  }
}

const newStack = new Stack;
console.log(newStack.checkIfEmpty());
newStack.addElement(5);
console.log(newStack.checkIfEmpty());
console.log(newStack.peek());
newStack.removeElement();
console.log(newStack.checkIfEmpty());
console.log(newStack.peek());

File: stack/ruby/stack.rb
class Stack
  def initialize
    @stack = []
    @top = -1
  end

  def add_element(item)
    @top += 1
    @stack << item
  end

  def check_if_empty
    @stack.empty?
  end

  def remove_element
    return 'Empty' if check_if_empty == true

    @top -= 1
    @stack.pop
  end

  def peek
    @stack[@top]
  end
end

new_stack = Stack.new
puts new_stack.peek
new_stack.add_element(5)
puts new_stack.peek
new_stack.remove_element
puts new_stack.peek
